relay-redis/src/real.rs

```diff
@@ -22,18 +26,28 @@ pub enum RedisError {
 
 enum ConnectionInner<'a> {
     Cluster(&'a mut redis::cluster::ClusterConnection),
+    MultiWrite {
+        primary: Box<ConnectionInner<'a>>,
+        secondaries: Vec<ConnectionInner<'a>>,
+    },
     Single(&'a mut redis::Connection),
 }
 
-/// A reference to a pooled connection from `PooledClient`.
-pub struct Connection<'a> {
-    inner: ConnectionInner<'a>,
-}
-
-impl ConnectionLike for Connection<'_> {
+impl ConnectionLike for ConnectionInner<'_> {
     fn req_packed_command(&mut self, cmd: &[u8]) -> redis::RedisResult<redis::Value> {
-        match self.inner {
+        match self {
             ConnectionInner::Cluster(ref mut con) => con.req_packed_command(cmd),
+            ConnectionInner::MultiWrite {
+                primary: primary_connection,
+                secondaries: secondary_connections,
+            } => {
+                let primary_result = primary_connection.req_packed_command(cmd);
+                for secondary_connection in secondary_connections.iter_mut() {
+                    secondary_connection.req_packed_command(cmd)?;
+                }
```

Comment:
Didn't we plan to execute multiple commands in parallel?

Reply:
Yes, I will investigate this.

Reply:
Yes, should have put this in the issue. We discussed it and decided to do it in a follow up.